Fix `<Text>` component text color value to correct color value for Link UI preview description text

getdave opened the pull request:
<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/repository-management.md#pull-requests. -->

<!-- Gutenberg's license is in the process of updating to be dual-licensed under the GPL and MPL. As part of that transition, all new contributions are dual-licensed. For more information, see: https://github.com/WordPress/gutenberg/blob/trunk/LICENSE.md -->

## Description
In https://github.com/WordPress/gutenberg/pull/35833#issuecomment-948592797 @jasmussen noticed we are using the wrong text color in the description text of the Link UI preview.

This was because the `<Text>` component uses black and not darkGray from the available color values.

This PR corrects the underlying Text component from `#000` to `$gray-900`. This has the knock on effect of fixing the LinkControl preview.

## How has this been tested?

1. Create link.
2. Click on link to show preview.
3. Inspect link description text.
4. Check color is `$gray-900` (`#1e1e1e`).
5. Check that color is applied via cascade inheriting from the styles on the `<Text>` component (as opposed to any styles applied directly to `<LinkControl>`).

## Screenshots <!-- if applicable -->
<img width="1051" alt="Screen Shot 2021-10-25 at 10 08 08" src="https://user-images.githubusercontent.com/444434/138667884-b0ba83a8-09dd-4d20-aebd-41edd17b79d1.png">



## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [ ] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [ ] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [ ] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->

jasmussen commented:
Very fast work, thank you! And it solves the problem:
<img width="459" alt="Screenshot 2021-10-22 at 09 06 40" src="https://user-images.githubusercontent.com/1204802/138409040-c8a86cb4-96c2-4865-a3d2-f3e72b1c90ee.png">

This PR specifies the right color. But it also overrides an existing black color, which I'd challenge to not be useful in the first place:

<img width="292" alt="Screenshot 2021-10-22 at 09 06 51" src="https://user-images.githubusercontent.com/1204802/138409089-6dd198e0-0e3f-4334-bbb2-a06fb9941259.png">

As it turns out, we do need to provide a baseline color for the component, as even if we were to remove that black color, the text would inherit all the way back to wp-admin styles. But I was thinking something along the lines of adding the color to `.components-popover__content`, so all text inside would be born with the right color.

My line of thinking is that the text component should be agnostic about what color it offers, and inherit that of its surroundings. That may be incorrect?

In any case, the black color appears to come from https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/text/styles.js#L12. A different fix might be to assign the gray-900 variable there?

Thanks again!

getdave commented:
> In any case, the black color appears to come from https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/text/styles.js#L12. A different fix might be to assign the gray-900 variable there?

Yeh I guess that could work? I'm going to ping someone working on components to try and get some insight into the decisions there.

Are you saying `$gray-900` is a standard black color for Gutenberg in general? If so then I would assume it's ok to apply it there.

jasmussen commented:
Marco is AFK these days, but perhaps @mirka can provide input?

> Are you saying $gray-900 is a standard black color for Gutenberg in general? If so then I would assume it's ok to apply it there.

Yes, for the current visual setup, we shouldn't actually use true black for anything but drop shadows and edgecases.

lynnbgriffith commented:
👍🏻 In the original context in G2 the Text component was assigned `ui.color.text`. In general, I agree that it makes sense for that component to use the default text color for the UI by default, in this case assigning `$gray-900` there.

getdave committed:
Use correct text color in Text component

See https://github.com/WordPress/gutenberg/pull/35851#issuecomment-949626157

getdave commented:
@jasmussen @griffbrad I've updated the `<Text>` component to use the correct color option to get the `#e1e1e1` color. This is now being inherited by `<LinkControl>` and so I've removed the original commit which added the color directly onto LinkControl.

Is this good to 🚢 ?

jasmussen reviewed (approved):
"darkGray.primary" suggests to me we need a refreshed naming scheme, ideally one that can either inspire/upgrade the existing SCSS one, or just plain adopt it:

```
$black: #000;			// Use only when you truly need pure black. For UI, use $gray-900.
$gray-900: #1e1e1e;
$gray-800: #2f2f2f;
$gray-700: #757575;		// Meets 4.6:1 text contrast against white.
$gray-600: #949494;		// Meets 3:1 UI or large text contrast against white.
$gray-400: #ccc;
$gray-300: #ddd;		// Used for most borders.
$gray-200: #e0e0e0;		// Used sparingly for light borders.
$gray-100: #f0f0f0;		// Used for light gray backgrounds.
$white: #fff;
```

But that's a separate issue.

Another solution is `color: inherit;` — I'm actualy not sure why we need to bake colors into the Text component at all. What if it's used in dark mode?

I'll give this one a green check, because the component currently ships with the wrong color, and this rectifies that. But I would appreciate if @griffbrad could also just sanity check this one, especially whether we can move straight on to `inherit` or not.

lynnbgriffith commented:
I’m hesitant to jump straight to `inherit` because I think it merits wider testing. In some cases, it’s unclear what we’d be inheriting from and that could lead to unexpected results, whereas just changing the default to `$gray-900` will be straightforward and predictable. I agree overall, though, that `inherit` is the ideal endpoint. Will discuss with other components folks and plan the next steps.

getdave commented:
@griffbrad We can always merge this fix now and look to refactor to `inherit` later if that requires wider discussion. I'll wait to hear.

lynnbgriffith commented:
Yeah. I think it’s totally fine to proceed with this fix and then revisit `inherit` separately.

getdave committed:
Update snapshots with amended color value

getdave committed:
Update other snapshot tests
